finagle-netty4/src/main/scala/com/twitter/finagle/netty4/channel/ChannelStatsHandler.scala

```diff
@@ -99,13 +101,16 @@ private[netty4] class ChannelStatsHandler(statsReceiver: StatsReceiver)
 
   override def channelInactive(ctx: ChannelHandlerContext) {
     closeChans.incr()
-    val channelStats = ctx.attr(ConnectionStatsKey).get
 
+    val channelStats = ctx.attr(ConnectionStatsKey).get
     connectionReceivedBytes.add(channelStats.bytesRead.get)
     connectionSentBytes.add(channelStats.bytesWritten.get)
 
-    val elapsed = ctx.attr(ConnectionDurationKey).get()
-    connectionDuration.add(elapsed().inMilliseconds)
+    ctx.attr(ConnectionDurationKey).get match {
+      case null => // the connection didn't initialize
```

Comment:
i believe that this happens if there are errors in the write() call (which precedes channelActive in clients)